presto-main/src/main/java/com/facebook/presto/operator/aggregation/GenericAccumulatorFactory.java

```diff
@@ -563,7 +608,10 @@ public void addIntermediate(GroupByIdBlock groupIdsBlock, Block block)
         @Override
         public void evaluateIntermediate(int groupId, BlockBuilder output)
         {
-            throw new UnsupportedOperationException();
+            RowBlock rowBlock = (RowBlock) pageMap.get((long) groupId).build();
+            for (int i = 0; i < rowBlock.getPositionCount(); i++) {
+                output.appendStructureInternal(rowBlock, i);
+            }
```

Comment:
So this bit is causing some issues since we cant append multiple rows to a single groupId. That's why the code wasn't working earlier, I was only putting one block per group id so we were missing a bunch of rows in the final step.

Now, when I add multiple blocks for a single row, the page becomes malformed since there is an inconsistent position count between its blocks. I.e. this column has multiple rows for a single input group Id.

Been thinking of some alternatives today and maybe we should just serialize this and then write it as bytes but that definitely would have overhead. What do you think @highker ?

Reply:
I think one good place to understand how this code should behave is by reading `AccumulatorCompiler`. `generateGroupedEvaluateIntermediate` is the method to generate `evaluateIntermediate`. Looks like only one row is allowed per `groupId` and it needs to be serialized. So for that, you could squeeze all the rowBlocks into an array or something to make a single cell.

Reply:
Yeah, array is a good idea, let me try that. Since this BlockBuilder is of the type returned by `getIntermediateType`, I'm going to set this to ArrayType and see if that works.

Reply:
Okay almost have it - what's left is I need to figure out how to attach the group id block offsets correctly to the serialized array and it *should* work.

Reply:
Added a bit to rebuild the groupidsblock according to the nested rowblock array items